crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -525,3 +525,97 @@ fn test_verify_nonce_in_unsigned_tx() {
+
+        let transaction = get_invoke_dummy(Felt252Wrapper::ZERO);
+
+        let tx_sender = transaction.sender_address.into();
+        let tx_source = TransactionSource::InBlock;
+        let call = Call::invoke { transaction: transaction.into() };
+
+        assert!(Starknet::validate_unsigned(tx_source, &call).is_ok());
+
+        set_nonce::<MockRuntime>(&tx_sender, &Nonce(StarkFelt::from(1u64)));
+
+        assert_eq!(
+            Starknet::validate_unsigned(tx_source, &call),
+            Err(TransactionValidityError::Invalid(InvalidTransaction::Stale))
+        );
+
+        let storage_key = (tx_sender, StorageKey(PatriciaKey(StarkFelt::from(0u128))));
+
+        assert_eq!(Starknet::storage(storage_key), StarkFelt::from(0u128));
+    });
+}
+
+
+#[test]
+fn contract_function_call_should_fail() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+
+        let none_origin = RuntimeOrigin::none();
+
+        let account_addr = get_account_address(None, AccountType::V1(AccountTypeV1Inner::NoValidate));
+
+        let transaction_panic_class = get_contract_class("TransactionPanic.casm.json", 1);
+        let transaction_panic_class_hash =
+            Felt252Wrapper::from_hex_be("0x7d2bcb1df4970245665a19b23a4d3877eb86a661e8d98b89afc4531134b99f6").unwrap();
+        let transaction_panic_compiled_class_hash: Felt252Wrapper =
+            Felt252Wrapper::from_hex_be("0x1c02b663e928ed213d3a0fa206efb59182fa2ba41f5c204daa56c4a434b53e5").unwrap();
+
+        let mut transaction = DeclareTransactionV2 {
+            sender_address: account_addr.into(),
+            class_hash: transaction_panic_class_hash,
+            compiled_class_hash: transaction_panic_compiled_class_hash,
+            nonce: Felt252Wrapper::ZERO,
+            max_fee: u128::MAX,
+            signature: vec![],
+            offset_version: false,
+        };
+
+        let chain_id = Starknet::chain_id();
+        let transaction_hash = transaction.compute_hash::<<MockRuntime as Config>::SystemHash>(chain_id, false);
+        transaction.signature = sign_message_hash(transaction_hash);
+
+        assert_ok!(Starknet::validate_unsigned(
+            TransactionSource::InBlock,
+            &crate::Call::declare {
+                transaction: transaction.clone().into(),
+                contract_class: transaction_panic_class.clone()
+            },
+        ));
+
+        assert_ok!(Starknet::declare(none_origin, transaction.into(), transaction_panic_class.clone()));
+        assert_eq!(
+            Starknet::contract_class_by_class_hash(ClassHash::from(transaction_panic_class_hash)).unwrap(),
+            transaction_panic_class
+        );
+
+        // log account address
+        log::info!("Account address: {:?}", account_addr);
+
+        let deploy_transaction = InvokeTransactionV1{
+            sender_address: account_addr.into(),
+            signature: vec![],
+            nonce: Felt252Wrapper::ONE,
+            calldata: vec![
```

Comment:
Hey @apoorvsadana , getting error (ref - screenshot) while running the deploy transaction here. 

<img width="729" alt="Screenshot 2024-03-23 at 12 43 17 AM" src="https://github.com/keep-starknet-strange/madara/assets/26214362/3f3dd79b-c9eb-415b-a8f5-827354504b2f">

I have an intuition that this is happening because of calldata. Could you please have a look?
The calldata is inspired from the network log for contract deployment transaction here - <img width="605" alt="Screenshot 2024-03-23 at 12 47 35 AM" src="https://github.com/keep-starknet-strange/madara/assets/26214362/047fdad3-d593-4c36-a97a-942d30c07997">

Reply:
Can you try making the nonce as 2? Because the nonce was 1 for the declare txn

Also, are you sure this error message is for this test? Because the account you're using isn't an Argent account. The normal tests can have error messages if they are checking for failures so just confirming you're looking at the right one. You can test your specific case by specifying the test name after cargo test

Reply:
- I had tried with nonce as 2, but it is still failing. 
- Yes you are right that error was getting logged from a different test case, my bad there.

Below is the error that is being thrown from this test. In summary the transaction execution is failing. I am trying to figure out the correct calldata for deploy transaction because only then the tx will go through. If you have any leads here please let me know

<img width="723" alt="Screenshot 2024-03-23 at 3 53 50 PM" src="https://github.com/keep-starknet-strange/madara/assets/26214362/99c61289-0827-40c5-87d3-afd0199fc762">

Reply:
from the image you shared, the ngrok shows 8 elements in the calldata but your test case has only 7?

Reply:
Yeah I missed adding the point that -

I deployed the same contract multiple times and according to my observation:
- All the values in the calldata array remains the same except for the 6th (5th index: calldata[5]) element. 
- This value changes for every transaction and I am unable to figure out what that could be